### PR TITLE
Pin cmake 3.22.3

### PR DESCRIPTION
**Pull request by zengk95**
Addresses #1010

Searched for all usages of `install.*cmake`

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: manywheel/Dockerfile [hunk 1]
```diff
@@ -20,7 +20,7 @@ RUN wget http://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm &
     rm -f epel-release-latest-7.noarch.rpm
 
 # cmake
-RUN yum install -y cmake3 && \
+RUN yum install -y cmake3-3.22.3 && \
```
Comment (malfet) [inline manywheel/Dockerfile:23]: Same as above
```suggestion
RUN yum install -y cmake3 && \
```

File: libtorch/Dockerfile [hunk 2]
```diff
@@ -42,7 +42,7 @@ ENV CUDA_HOME /usr/local/cuda
 FROM base as conda
 ADD ./common/install_conda.sh install_conda.sh
 RUN bash ./install_conda.sh && rm install_conda.sh
-RUN /opt/conda/bin/conda install -y cmake=3.14
+RUN /opt/conda/bin/conda install -y cmake=3.22.3
```
Comment (malfet) [inline libtorch/Dockerfile:45]: Let's stick with 3.14 as it allows us to validate that CMakeFile is correct at minspec
```suggestion
RUN /opt/conda/bin/conda install -y cmake=3.14
```

File: manywheel/conda_build.sh [hunk 3]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env bash
 set -e
 
-yum install -y wget git cmake
+yum install -y wget git cmake-3.22.3
```
Comment (malfet) [inline manywheel/conda_build.sh:4]:
```suggestion
yum install -y wget git cmake
```

**Review by malfet — CHANGES_REQUESTED** [review]
Do not pin version in `apt-get`/`yum install` command, as cmake=3.23 is not available there yet (which is probably reflected in CI failures)

**Comment by zengk95** [comment]
@malfet I removed those pins as suggested. Could you review again?

**Review by malfet — APPROVED** [review]
Thank you for the updates, looks good to me

**Comment by malfet** [comment]
Looks like it caused MacOS night jobs to fail, as cmake-3.22.3 is not available from Anaconda, is it: https://anaconda.org/anaconda/cmake

**Pull request merged**